protocols/secio/src/codec/mod.rs

```diff
@@ -109,12 +109,15 @@ pub fn full_codec<S>(
     encoding_hmac: Hmac,
     cipher_decoder: StreamCipher,
     decoding_hmac: Hmac,
+    remote_nonce: Vec<u8>
 ) -> FullCodec<S>
 where
     S: AsyncRead + AsyncWrite,
 {
     let encoder = EncoderMiddleware::new(socket, cipher_encoding, encoding_hmac);
-    DecoderMiddleware::new(encoder, cipher_decoder, decoding_hmac)
+    let mut decoder = DecoderMiddleware::new(encoder, cipher_decoder, decoding_hmac);
+    decoder.remote_nonce(remote_nonce);
```

Comment:
This piece looks a bit odd, seems like the `remote_nonce` could be part of the `new()` call?